testSrc/unit/io/flutter/editor/FlutterEditorAnnotatorTest.java

```diff
@@ -0,0 +1,152 @@
+ * found in the LICENSE file.
+ */
+package io.flutter.editor;
+
+import com.intellij.codeInsight.daemon.impl.AnnotationHolderImpl;
+import com.intellij.lang.annotation.Annotation;
+import com.intellij.lang.annotation.AnnotationSession;
+import com.intellij.lang.annotation.HighlightSeverity;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.impl.source.tree.LeafPsiElement;
+import com.jetbrains.lang.dart.psi.DartCallExpression;
+import com.jetbrains.lang.dart.psi.DartNewExpression;
+import com.jetbrains.lang.dart.psi.DartReferenceExpression;
+import io.flutter.AbstractDartElementTest;
+import io.flutter.dart.DartSyntax;
+import org.junit.Test;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+
+public class FlutterEditorAnnotatorTest extends AbstractDartElementTest {
+  @Test
+  public void locatesColorReference() throws Exception {
+    run(() -> {
+      final PsiElement testIdentifier = setUpDartElement("main() { Colors.blue; }", "Colors", LeafPsiElement.class);
+      final DartReferenceExpression element = DartSyntax.findEnclosingReferenceExpression(testIdentifier);
+      assert element != null;
+
+      final FlutterEditorAnnotator annotator = new FlutterEditorAnnotator();
+      final AnnotationSession annotationSession = new AnnotationSession(testIdentifier.getContainingFile());
+      final AnnotationHolderImpl annotationHolder = new AnnotationHolderImpl(annotationSession);
+
+      annotator.annotate(element, annotationHolder);
+
+      assertTrue(annotationHolder.hasAnnotations());
+      assertEquals(1, annotationHolder.size());
+
+      final Annotation annotation = annotationHolder.get(0);
+      assertEquals(HighlightSeverity.INFORMATION, annotation.getSeverity());
+    });
+  }
+
+  @Test
+  public void locatesIconReference() throws Exception {
+    run(() -> {
+      final PsiElement testIdentifier = setUpDartElement("main() { Icons.access_alarm; }", "Icons", LeafPsiElement.class);
+      final DartReferenceExpression element = DartSyntax.findEnclosingReferenceExpression(testIdentifier);
+      assert element != null;
+
+      final FlutterEditorAnnotator annotator = new FlutterEditorAnnotator();
+      final AnnotationSession annotationSession = new AnnotationSession(testIdentifier.getContainingFile());
+      final AnnotationHolderImpl annotationHolder = new AnnotationHolderImpl(annotationSession);
+
+      annotator.annotate(element, annotationHolder);
+
+      assertTrue(annotationHolder.hasAnnotations());
+      assertEquals(1, annotationHolder.size());
+
+      final Annotation annotation = annotationHolder.get(0);
+      assertEquals(HighlightSeverity.INFORMATION, annotation.getSeverity());
+    });
+  }
+
+  @Test
+  public void locatesColorCtor() throws Exception {
+    run(() -> {
+      final PsiElement testIdentifier = setUpDartElement("main() { Color(0xFFE3F2FD); }", "Color", LeafPsiElement.class);
```

Comment:
Could be worth it to add matching cases with
`const Color(0xFFE3F2FD)`

`const [Color(0xFFE3F2FD])`
and 
`new Color(0xFFE3F2FD)`
and
cases with excess whitespace
`new    Color   (
0xFFE3F2FD
)`
to make sure that the matching based on whitespace differences isn't fragile.

Reply:
Adding `const [Color(0xFFE3F2FD])` and `const Color ( 0xFFE3F2FD )` (which did catch an issue). The `new` cases are currently not supported, and, I'm not 100% sure how important it is to support them (most or all user references are of the form `Color.fooColor`; the const Color(0xFFF)` references are generally just in `package:flutter/material/colors.dart`).